patches/minecraft/net/minecraft/world/server/ServerWorld.java.patch

```diff
@@ -73,15 +73,17 @@
        }
  
        if (this.func_72912_H().func_76093_s() && this.func_175659_aa() != Difficulty.HARD) {
-@@ -280,7 +291,7 @@
+@@ -280,8 +291,8 @@
        })) {
           this.field_73068_P = false;
           if (this.func_82736_K().func_223586_b(GameRules.field_223607_j)) {
 -            long l = this.field_72986_A.func_76073_f() + 24000L;
+-            this.func_72877_b(l - l % 24000L);
 +            long l = this.func_72820_D() + 24000L;
-             this.func_72877_b(l - l % 24000L);
++            this.func_72877_b(net.minecraftforge.event.ForgeEventFactory.onSleepFinished(this, l - l % 24000L));
```

Comment:
Would this cause a problem if the resulting time is before the current time? (I'm not 100% sure how time is handled in MC). If that is the case, would it not make more sense to just return a time-of-day, rather than an absolute time?

Reply:
It was more of a question than a request for change.
Can you still sleep for less than a night with the new approach? If not it would be best to revert that commit.

Might be better to wait for someone who knows more about it to comment before making further changes.

Reply:
now it adds the time to the current daytime (adds min 0 ticks, max 1 day)

^^should be better than my first approach